### PR TITLE
Drag

### PR DESCRIPTION
**Pull request by acaldas**
https://github.com/ripe-tech/ripe-core/issues/2228

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ As a starting point, you need to provide the base `url` of the server where the
 You may also pass an `options` map to override parameters like `currency` and `country`, which are 'EUR' and 'US' respectively by default.
 
 ```javascript
-var ripe = new Ripe(url, brand, model, variant, parts, {
+var ripe = new Ripe(url, brand, model, variant, parts, frames, {
```
Comment (hugo-gomes) [inline README.md]: what is the `frames` parameter?

Comment (hugo-gomes) [inline README.md]: move the `parts` to be within the options parameter

File: src/js/base.js [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    request.addEventListener("load", function() {
+        var isValid = this.status === 200;
+        var result = JSON.parse(this.responseText);
+        callback.call(context, isValid ? result.combinations : null);
+    });
+    request.open("GET", combinationsURL);
+    request.send();
+};
+
+Ripe.prototype._getImageURL = function(frame, parts, brand, model, variant, engraving, options) {
```
Comment (hugo-gomes) [inline src/js/base.js]: move all the requests related code to a new file `remote.part.js`

File: src/js/bindDrag.js [hunk 3]
```diff
@@ -0,0 +1,510 @@
+Ripe.prototype.bindDrag = function(target, frames, size, maxSize, options) {
```
Comment (hugo-gomes) [inline src/js/bindDrag.js]: rename this file to `bind_drag.part.js`

File: src/js/bindDrag.js [hunk 4]
```diff
@@ -0,0 +1,510 @@
+Ripe.prototype.bindDrag = function(target, frames, size, maxSize, options) {
+    // validates that the provided target element is a
+    // valid one and if that's not the case returns the
+    // control flow immediately to the caller
+    if (!target) {
+        return;
+    }
+
+    // sets sane defaults for the optional parameters
+    size = size || 1000;
```
Comment (hugo-gomes) [inline src/js/bindDrag.js]: this hardcoded values must be option's default values

File: src/js/bindDrag.js [hunk 5]
```diff
@@ -0,0 +1,510 @@
...
+        // retrieves the last recorded mouse position
+        // and the current one and calculates the
+        // drag movement made by the user
+        var child = element.querySelector("*:first-child");
+        var referenceX = element.dataset.referenceX;
+        var referenceY = element.dataset.referenceY;
+        var mousePosX = element.dataset.mousePosX;
+        var mousePosY = element.dataset.mousePosY;
+        var base = element.dataset.base;
+        var rate = rate || 40;
```
Comment (hugo-gomes) [inline src/js/bindDrag.js]: this should be defined within the options

File: README.md (hunk 1 above)
Comment (hugo-gomes) [inline README.md]: what is the frames parameter?

File: src/js/bindDrag.js [hunk 6]
```diff
@@ -0,0 +1,508 @@
+Ripe.prototype.bindDrag = function(target, size, maxSize, options) {
```
Comment (hugo-gomes) [inline src/js/bindDrag.js]: rename this file to `bind_drag.part.js`

File: src/js/base.js [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+    }
+
+    this.ready && this._runCallbacks("update");
+
+    this.ready && this.getPrice(function(value) {
+        this._runCallbacks("price", value);
+    });
+};
+
+Ripe.prototype.getPrice = function(callback) {
```
Comment (hugo-gomes) [inline src/js/base.js]: move all remote requesting related code to a new `remote.part.js`

File: src/python/ripe_demo/static/js/main.js [hunk 8]
```diff
@@ -10,8 +10,12 @@ window.onload = function() {
     var parts = [];
     var partsMap = {};
     var index = 0;
+    var frames = {
```
Comment (hugo-gomes) [inline src/python/ripe_demo/static/js/main.js]: this mustn't be hardcode.

use the `/api/brands/<str:brand>/models/<str:model>/config` endpoint

File: src/js/base.part.js [hunk 9]
```diff
@@ -158,45 +158,6 @@ Ripe.prototype.update = function(price) {
     });
 };
 
-Ripe.prototype.getPrice = function(callback) {
```
Comment (hugo-gomes) [inline src/js/base.part.js]: move `_getImageURL`, `_getPriceURL`, `_getDefaultsURL`, `_getCombinationsURL`, `_getQuery` also

File: src/python/ripe_demo/static/js/main.js (hunk 8 above)
Comment (acaldas) [reply src/python/ripe_demo/static/js/main.js]: The available frames of a product can't be inferred with information currently provided:

```
"frames": 24,
"faces": ["side", "top"]
```

File: src/js/ripe.js [hunk 10]
```diff
@@ -693,6 +705,26 @@ Ripe.prototype._updateDrag = function(target, position, animate, single, callbac
...
 
+Ripe.prototype.getFrames = function(callback) {
+    if (this.config === undefined) {
+        this.getConfig(function(config) {
+            this.config = config;
+            this.getFrames(callback);
+        });
+    }
+
+    var frames = {};
```
Comment (hugo-gomes) [inline src/js/ripe.js]: update the frames references on current documentation

Comment (acaldas) [reply src/js/ripe.js]: Done on https://github.com/ripe-tech/ripe-sdk/pull/3

**Pull request closed without merging**